README.md

```diff
@@ -34,12 +34,93 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Getting Started
+
+To get started, configure `Cargo.toml` to include `tracing` and `tracing-subscriber` as dependencies. `tracing` is the core library, while `tracing-subscriber` is the part that implements `Subscriber`, which will record the trace. Take a look at the [Project layout](#Project layout) section.
+
+```
+[dependencies]
+tracing = "0.1.5"
+tracing-subscriber = "0.1.5"
+```
+
+Next, in our `main.rs` file, we will start by adding the libraries
+
+```
+use tracing::{debug, info, instrument};
+use tracing_subscriber::fmt;
+```
+
+For this simple example, we will create a function that prints sequence of numbers and we will instrument that cod to enable tracing. Tracing is enabled by initiating a span and the `#[instrument]` attribute, implicitly create a span for that function.
+
+```
+// using instrument to setup a span and skip to exclude args
+#[instrument(skip(_too_long_string))]
+fn count(nums: i32, _too_long_string: &str) -> Vec<i32> {
+    let mut seq = vec![];
+
+    for n in 0..=nums {
+        // logging n in structured form
+        debug!(n, "Next Number");
+        seq.push(n);
+    }
+
+    seq
+}
+```
+
+Then, in our `main()` function, we can create a subscriber, that records the trace. The subscriber employs an environment filter that we can utilize to filter the level of tracing message.
+
+```
+fn main() {
+  // this creates subscriber with an environment filter
```

Comment:
Personally, I'd prefer to start by introducing how to create and set a subscriber with the default filter
```rust
let subscriber = fmt::Subscriber::default();
tracing::subscriber::with_default(subscriber, || {
    // ...
});
```
and then introduce customizing the filter _after_ we demonstrate the default...